.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,50 @@
+---
+## Workflow to lint and test
+name: pull-request
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    env:
+      JENKINS_URL: https://beats-ci.elastic.co/
```

Comment:
What's this being used for?

Reply:
for linting the Jenkins pipelines, there is a pre-commit validation we provide in https://github.com/elastic/apm-pipeline-library/blob/b23be9e84d84f12f8586e73a0490a07fcfd2f221/.pre-commit-hooks.yaml#L22-L29

Eventually to be removed once it's not Jenkins based